R/selectors.R

```diff
@@ -78,30 +78,32 @@ html_elements <- function(x, css, xpath) {
 
 #' @export
 html_elements.default <- function(x, css, xpath) {
-  xml2::xml_find_all(x, make_selector(css, xpath))
+  xpath <- make_selector(css, xpath)
+  xml2::xml_find_all(x, xpath)
 }
 
 #' @export
 html_element.default <- function(x, css, xpath) {
-  xml2::xml_find_first(x, make_selector(css, xpath))
+  xpath <- make_selector(css, xpath)
+  xml2::xml_find_first(x, xpath)
 }
 
-make_selector <- function(css, xpath) {
-  if (missing(css) && missing(xpath))
-    stop("Please supply one of css or xpath", call. = FALSE)
-  if (!missing(css) && !missing(xpath))
-    stop("Please supply css or xpath, not both", call. = FALSE)
+make_selector <- function(css, xpath, call = rlang::caller_env()) {
+  switch(
```

Comment:
I'd suggest keeping the basic flow of the function the same, just calling `check_exclusive()` (without the `rlang::`) first. I'd suggest you leave the string checking messages as is in this PR, as those would be better handled with the standalone rlang input checkers.